packages/lab/src/dialog/Dialog.tsx

```diff
@@ -89,49 +114,39 @@ export const Dialog = forwardRef<HTMLDivElement, DialogProps>(function Dialog(
     [dialogId, status]
   );
 
+  if (!open && !showComponent) return null;
+
   return (
-    <FloatingPortal>
-      {/* The provider is needed to support the use case where an app has nested modes. The element that is portalled needs to have the same style as the current scope */}
-      <SaltProvider>
-        {showComponent && (
-          <FloatingOverlay className={withBaseName("overlay")} lockScroll>
-            <FloatingFocusManager
-              context={context}
-              modal
-              initialFocus={initialFocus}
-            >
-              <DialogContext.Provider value={contextValue}>
-                <div
-                  id={dialogId}
-                  className={clsx(
-                    withBaseName(),
-                    {
-                      [withBaseName("enterAnimation")]: open,
-                      [withBaseName("exitAnimation")]: !open,
-                      [withBaseName("withStatus")]: status,
-                      [withBaseName(status as string)]: status,
-                    },
-                    className
-                  )}
-                  onAnimationEnd={() => {
-                    if (!open && showComponent) {
-                      setShowComponent(false);
-                    }
-                  }}
-                  ref={floatingRef}
-                  aria-labelledby={`${dialogId}-heading`}
-                  aria-describedby={`${dialogId}-description`}
-                  aria-modal="true"
-                  {...getFloatingProps()}
-                  {...rest}
-                >
-                  {children}
-                </div>
-              </DialogContext.Provider>
-            </FloatingFocusManager>
-          </FloatingOverlay>
-        )}
-      </SaltProvider>
-    </FloatingPortal>
+    <Scrim>
+      <DialogContext.Provider value={contextValue}>
+        <FloatingComponent
+          open={open}
+          aria-modal="true"
+          ref={floatingRef}
+          focusManagerProps={{
+            context: context,
+          }}
+          className={clsx(
+            withBaseName(),
+            withBaseName(size, currentbreakpoint),
+            {
+              [withBaseName("enterAnimation")]: open,
+              [withBaseName("exitAnimation")]: !open,
+              [withBaseName(status as string)]: status,
+            },
+            className
+          )}
+          onAnimationEnd={() => {
```

Comment:
Did you test this with animations turned off?

Reply:
With animations turned off the Dialog won't close. The alternative is to use the setTimeOut() function to close the dialog but from my reading this can be temperamental

Reply:
To be resolved in next sprint when promoting Dialog to Core